File: rest-service/queue/src/main/java/mb/lib/extern/model/QueueJobStatus.java
package mb.lib.extern.model;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum QueueJobStatus
{
  Completed,
  Errored,
  Queued,
  InProgress;

  @JsonCreator
  public static QueueJobStatus fromString(String value) {
    var status = value.toLowerCase();

    if ("completed".equals(status))
      return QueueJobStatus.Completed;

    if (status.contains("fail"))
      return QueueJobStatus.Errored;

    if ("grabbed".equals(status) || "claimed".equals(status))
      return QueueJobStatus.InProgress;

    if ("waiting".equals(status))
      return QueueJobStatus.Queued;

    throw new IllegalArgumentException(String.format("Invalid JobStatus value \"%s\".", value));
  }

  @Override
  public String toString() {
    return name();
  }
}

File: rest-service/src/main/kotlin/mb/lib/queue/model/QueueJobStatus.kt
package mb.lib.queue.model

import com.fasterxml.jackson.annotation.JsonCreator
import java.util.*

enum class QueueJobStatus {
  Completed, Errored, Queued, InProgress;

  override fun toString() = name

  companion object {
    @JsonCreator
    fun fromString(value: String): QueueJobStatus {
      val status = value.lowercase(Locale.getDefault())
      return when {
        "completed" == status                      -> Completed
        status.contains("fail")              -> Errored
        "grabbed" == status || "claimed" == status -> InProgress
        "waiting" == status                        -> Queued
        else                                       -> throw IllegalArgumentException("Invalid JobStatus value \"$value\".")
      }
    }
  }
}